homeassistant/components/zamg/config_flow.py

```diff
@@ -33,25 +34,34 @@ async def async_step_user(
             self._client.session = async_get_clientsession(self.hass)
 
         if user_input is None:
-            closest_station_id = await self._client.closest_station(
-                self.hass.config.latitude,
-                self.hass.config.longitude,
-            )
-            LOGGER.debug("config_flow: closest station = %s", closest_station_id)
-            stations = await self._client.zamg_stations()
-            user_input = {}
-
-            schema = vol.Schema(
-                {
-                    vol.Required(CONF_STATION_ID, default=closest_station_id): vol.In(
-                        {
-                            station: f"{stations[station][2]} ({station})"
-                            for station in stations
-                        }
-                    )
-                }
-            )
-            return self.async_show_form(step_id="user", data_schema=schema)
+            try:
+                stations = await self._client.zamg_stations()
+                closest_station_id = await self._client.closest_station(
+                    self.hass.config.latitude,
+                    self.hass.config.longitude,
+                )
+                LOGGER.debug("config_flow: closest station = %s", closest_station_id)
```

Comment:
Please only wrap the lines that can raise in the try... except block. Use an `else:` block if needed.